src/rest-server/deploy/legacy_user_migrate.py

```diff
@@ -138,6 +138,60 @@ def secret_data_prepare_v2_group(self, groupname):
         post_data_dict['data'] = group_dict
         return post_data_dict
 
+    def load_v2_groups(self):
+        ns_pai_group_list = self.list_all_secrets_from_namespace(self.secret_ns_group_v2)
+        decode_group_list = []
+        vc_set = set()
+        for group in ns_pai_group_list:
+            meta_dict = dict()
+            meta_dict['name'] = bytes.fromhex(group.metadata.name).decode('utf-8')
+            group_dict = {
+                'groupname': str(base64.b64decode(group.data['groupname'].encode('utf-8')), 'utf-8'),
+                'description': str(base64.b64decode(group.data['description'].encode('utf-8')), 'utf-8'),
+                'externalName': str(base64.b64decode(group.data['externalName'].encode('utf-8')), 'utf-8'),
+                'extension': json.loads(str(base64.b64decode(group.data['extension'].encode('utf-8')), 'utf-8')),
+            }
+            decode_group_list.append({
+                'metadata': meta_dict,
+                'data': group_dict,
+            })
+            if group_dict['extension'].get('groupType') == 'vc':
+                vc_set.add(group_dict['groupname'])
+        return decode_group_list, vc_set
+
+    def convert_v2_group(self, data_dict, all_vcs):
+        extension_dict = data_dict['data']['extension']
+        if 'acls' in extension_dict:
+            return data_dict, False
+
+        extension_dict['acls'] = {
+            'admin': False,
+            'virtualClusters': [],
+        }
+        if 'groupType' in extension_dict:
+            if extension_dict['groupType'] == 'admin':
+                extension_dict['acls']['admin'] = True
+                extension_dict['acls']['virtualClusters'].extend(all_vcs)
+            elif extension_dict['groupType'] == 'vc':
+                extension_dict['acls']['virtualClusters'].append(data_dict['data']['groupname'])
+
+        return data_dict, True
+
+    def update_v2_group(self, data_dict):
+        post_data_dict = {
+            'metadata': {
+                'name': data_dict['metadata']['name'].encode('utf-8').hex()
```

Comment:
In tag1, I think it's unnecessary to do decode and then encode here. Why not directly use the value from k8s?

Reply:
They are different funcs, we should make them more genernal. In other words, this func shouldn't assume input is a partly encoded object.